visualization/app/codeCharta/state/store/metricData/metricData.service.spec.ts

```diff
@@ -0,0 +1,97 @@
+import "../../state.module"
+import { IRootScopeService } from "angular"
+import { MetricDataService } from "./metricData.service"
+import { EdgeMetricDataService } from "./edgeMetricData/edgeMetricData.service"
+import { setEdgeMetricData } from "./edgeMetricData/edgeMetricData.actions"
+import { setNodeMetricData } from "./nodeMetricData/nodeMetricData.actions"
+import { getService, instantiateModule } from "../../../../../mocks/ng.mockhelper"
+import { EDGE_METRIC_DATA, METRIC_DATA, VALID_EDGE, withMockedEventMethods } from "../../../util/dataMocks"
+import { NodeMetricDataService } from "./nodeMetricData/nodeMetricData.service"
+import { StoreService } from "../../store.service"
+import { setEdges } from "../fileSettings/edges/edges.actions"
+
+describe("MetricDataService", () => {
+	let metricDataService: MetricDataService
+	let storeService: StoreService
+	let $rootScope: IRootScopeService
+
+	beforeEach(() => {
+		restartSystem()
+		rebuildService()
+		withMockedEventMethods($rootScope)
+	})
+
+	function restartSystem() {
+		instantiateModule("app.codeCharta.state")
+
+		$rootScope = getService<IRootScopeService>("$rootScope")
+		storeService = getService<StoreService>("storeService")
+	}
+
+	function rebuildService() {
```

Comment:
```suggestion
	function resetMetricDataService() {
```

Reply:
When creating redux properties or a subreducer, we have to create a lot of files and adapt some files. This is tedious. That's why we use plop to auto-generate those for us. I'd rather do this in another PR and update the templates accordingly if this is wished. It just takes time.

Reply:
SGTM